test/e2e/framework/network/utils.go

```diff
@@ -212,7 +245,11 @@ func (config *NetworkingTestConfig) diagnoseMissingEndpoints(foundEndpoints sets
 func (config *NetworkingTestConfig) EndpointHostnames() sets.String {
 	expectedEps := sets.NewString()
 	for _, p := range config.EndpointPods {
-		expectedEps.Insert(p.Name)
+		if config.EndpointsHostNetwork {
+			expectedEps.Insert(p.Spec.NodeSelector["kubernetes.io/hostname"])
```

Comment:
Should this be checking `p.Spec.NodeName` or `p.Labels` instead? Checking node selector to extract the hostname seems weird.

Reply:
it is the weird it is configured in the manifest, you can see the code here 
https://github.com/kubernetes/kubernetes/blob/b61faa388c0d03fe2ae99b5b36e1d6f4d71eba29/test/e2e/framework/network/utils.go#L578-L580

so I took it directly, totally open to suggestions

Reply:
this is taking it from the configuration object, it does not read it from the API ,so it seems the only place to take it